File: palindromo.java
import java.util.Scanner;

public class palindromo{

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        String string = scanner.nextLine();

        while(!"FIM".equals(string)){
            if(ehpalindromo(string)){
            System.out.println("SIM");
            }
            else System.out.println("NAO");

            string = scanner.nextLine();
        } 

        scanner.close();
    }

    private static boolean ehpalindromo(String string){

        boolean flag = true;

        int i = 0;

        while(i < string.length()/2 && flag){
            if(string.charAt(i) != string.charAt(string.length() -1 -i) ) {
                flag = false;
            }
            i++;
        }

        return flag;
    }


}

File: palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

bool ehpalindromo(char string[])
{

    bool flag = true;

    int i = 0, j = strlen(string) - 1;

    while (i <= j)
    {
        if (string[i] != string[j])
        {
            flag = false;
            i = j;
        }
        else
        {
            i++;
            j--;
        }
    }

    return flag;
}

int main()
{

    char string[500];

    scanf(" %[^\n]", string);

    while(strcmp(string , "FIM") != 0){
        if(ehpalindromo(string)){
            printf("SIM\n");
        }else{
            printf("NAO\n");
        }

        scanf(" %[^\n]", string);
    }

    return 0;
}